test-suite/shared/src/test/scala/org/scalajs/testsuite/niobuffer/BufferFactory.scala

```diff
@@ -164,8 +164,11 @@ object BufferFactory {
     def allocBuffer(capacity: Int): BufferType =
       baseWrap(new Array[ElementType](capacity))
 
-    override def allocBuffer(pos: Int, limit: Int, capacity: Int): BufferType =
-      baseWrap(new Array[ElementType](capacity), pos, limit-pos)
+    override def allocBuffer(pos: Int, limit: Int, capacity: Int): BufferType = {
+      if (pos > limit)
+        throw new IllegalArgumentException
```

Comment:
Why is this change in the test suite? It should probably be somewhere in the `javalib/`, in whatever method is called by `baseWrap`. Otherwise we're not actually testing the implementation.